code/modules/mob/living/carbon/xenomorph/abilities.dm

```diff
@@ -165,33 +177,36 @@
 	to_chat(owner, "<span class='notice'>We will now spawn <b>[weed_choice]\s</b> when using the plant weeds ability.</span>")
 	update_button_icon()
 
-/datum/action/xeno_action/plant_weeds/update_button_icon()
+/datum/action/xeno_action/activable/plant_weeds/update_button_icon()
 	button.overlays.Cut()
 	button.overlays += image('icons/mob/actions.dmi', button, initial(weed_type.name))
 	return ..()
 
 //AI stuff
-/datum/action/xeno_action/plant_weeds/ai_should_start_consider()
+/datum/action/xeno_action/activable/plant_weeds/ai_should_start_consider()
 	return TRUE
 
-/datum/action/xeno_action/plant_weeds/ai_should_use(target)
+/datum/action/xeno_action/activable/plant_weeds/ai_should_use(target)
 	if(!can_use_action(override_flags = XACT_IGNORE_SELECTED_ABILITY))
 		return ..()
 	if(locate(/obj/effect/alien/weeds/node) in owner.loc) //NODE SPAMMMM
 		//There's already a node on this loc don't plant anything
 		return ..()
 	return TRUE
 
-/datum/action/xeno_action/plant_weeds/slow
-	cooldown_timer = 12 SECONDS
+/datum/action/xeno_action/activable/plant_weeds/ranged
+	max_range = 5
+	cooldown_timer = 5 SECONDS
 
-/datum/action/xeno_action/plant_weeds/slow/action_activate()
-	if(locate(/obj/effect/alien/weeds) in range(1, owner.loc))
-		return ..()
-	var/mob/living/carbon/xenomorph/hivemind/hiveminde = owner
-	hiveminde.forceMove(get_turf(hiveminde.core))
-	to_chat(hiveminde, span_xenonotice("We can't plant a node without weeds nearby, we've been moved back to our core."))
-	return fail_activate()
+/datum/action/xeno_action/activable/plant_weeds/ranged/can_use_ability(atom/A, silent = FALSE, override_flags)
+	var/area/a_area = get_area(A)
+	if(is_type_in_typecache(a_area, GLOB.wraith_no_incorporeal_pass_areas) && SSmonitor.gamestate == SHUTTERS_CLOSED)
+		to_chat(owner, span_warning("You cannot plant weeds here yet!"))
+		return FALSE
+	if(!owner.line_of_sight(get_turf(A)))
```

Comment:
probably make a better line of sight since theres a big range in here, for 1 or 2 or even 3 tiles is fine but after that its pretty bad, TG has some good helpers for that, like getline()